topic_metadata.go

```diff
@@ -120,27 +135,160 @@ func (t *Topic) parsePartitions(value []byte) (PartitionList, error) {
 	return result, nil
 }
 
+// marshalPartitions turns a PartitionList into the JSON representation
+// to be stored in Zookeeper.
+func (t *Topic) marshalPartitions(partitions PartitionList) ([]byte, error) {
+	tm := topicMetadata{Version: 1, Partitions: make(map[string][]int32, len(partitions))}
+	for _, part := range partitions {
+		tm.Partitions[fmt.Sprintf("%d", part.ID)] = part.Replicas
+	}
+	return json.Marshal(tm)
+}
+
+// generatePartitionAssignments creates a partition list for a topic. The primary replica for
+// each partition is assigned in a round-robin fashion starting at a random broker.
+// Additional replicas are assigned to subsequent brokers to ensure there is no overlap
```

Comment:
At some point, we can probably make this algorithm a bit more intelligent, but for now I think this is a good place to start :+1: